apps/vscode/src/main.ts

```diff
@@ -125,9 +126,44 @@ export async function activate(context: vscode.ExtensionContext) {
 
   // activate providers common to browser/node
   activateCommon(context, host, engine, commands);
+
+  // if positron
+  setEditorOpener();
+
+  vscode.workspace.onDidChangeConfiguration(async (event) => {
+    if (event.affectsConfiguration('quarto.defaultEditor')) {
+      setEditorOpener();
+    }
+  });
+
+  const documentOpenHandler = vscode.window.onDidChangeActiveTextEditor(async () => {
```

Comment:
I'm tempted to suggest that we move both of these listeners to `VisualEditorProvider.register()`. It gets called by `activateEditor()` above so it would still run at the right time, but:
- Both of these are about the editor, so putting it in `activateEditor()` makes sense to me
- In `register()` we already do things like register listeners for `workspace.onDidChangeTextDocument()`, so to me these are just "one more" editor related listeners and they belong beside the others.

Reply:
This makes a lot of sense to me! Moved 👍